src/stripeSamples.ts

```diff
@@ -0,0 +1,302 @@
+        : 'The sample was successfully created, but we could not set the API keys in the .env file. Please set them manually.';
+
+      await this.promptOpenFolder(postInstallMessage, clonePath);
+    } catch (e) {
+      vscode.window.showErrorMessage(`Cannot create Stripe sample: ${e.message}`);
+    }
+  };
+
+  /**
+   * Get a list of Stripe Samples items to show in a quick pick menu.
+   */
+  private getQuickPickItems = async () => {
+    const rawSamples = await new Promise<SamplesListResponse.SampleData[]>((resolve, reject) => {
+      this.daemonClient?.samplesList(new SamplesListRequest(), (error, response) => {
+        if (error) {
+          reject(error);
+        } else if (response) {
+          resolve(response.getSamplesList());
+        }
+      });
+    });
+
+    // alphabetical order
+    rawSamples.sort((a, b) => {
+      if (a.getName() < b.getName()) {
+        return -1;
+      }
+      if (a.getName() > b.getName()) {
+        return 1;
+      }
+      return 0;
+    });
+
+    const samplesQuickPickItems: SampleQuickPickItem[] = rawSamples.map((s) => {
+      return {
+        label: `$(repo) ${s.getName()}`,
+        detail: s.getDescription(),
+        sampleData: {
+          name: s.getName(),
+          url: s.getUrl(),
+        },
+      };
+    });
+
+    return samplesQuickPickItems;
+  };
+
+  /**
+   *  Get the available configs for this sample.
+   */
+  private getConfigsForSample(sampleName: string): Promise<SampleConfigsResponse.Integration[]> {
+    const request = new SampleConfigsRequest();
+    request.setSampleName(sampleName);
+
+    return new Promise((resolve, reject) => {
+      this.daemonClient?.sampleConfigs(request, (error, response) => {
+        if (error) {
+          reject(error);
+        } else if (response) {
+          resolve(response.getIntegrationsList());
+        }
+      });
+    });
+  }
+
+  /**
+   * Ask for which sample to clone.
+   */
+  private promptSample = async (): Promise<SampleQuickPickItem | undefined> => {
+    const selectedSample = await vscode.window.showQuickPick(this.getQuickPickItems(), {
+      matchOnDetail: true,
+      placeHolder: 'Select a sample to clone',
+    });
+    return selectedSample;
+  };
+
+  /**
+   * Ask for which integration to copy for this sample.
+   */
+  private promptIntegration = async (
+    sample: SampleQuickPickItem,
+  ): Promise<SampleConfigsResponse.Integration | undefined> => {
+    let integrations: SampleConfigsResponse.Integration[] | undefined;
+
+    const getIntegrationNames = async (sampleName: string) => {
+      integrations = await this.getConfigsForSample(sampleName);
+      return integrations.map((i) => i.getIntegrationName());
+    };
+
+    const selectedIntegrationName = await vscode.window.showQuickPick(
+      await getIntegrationNames(sample.sampleData.name),
+      {
+        placeHolder: 'Select an integration',
+      },
+    );
+
+    if (!integrations) {
+      return;
+    }
```

Comment:
Should we return before we show the quick pick if there's no integration?

Reply:
That makes sense to me. Fixing.